### PR TITLE
docs(zh): translate /config/lib to Chinese

### PR DESCRIPTION
**Pull request by fi3ework**
## Checklist

- [ ] Tests updated (or not required).
- [ ] Documentation updated (or not required).

### CODE REVIEW

File: website/docs/en/config/lib/banner.mdx [hunk 1]
```diff
@@ -16,7 +16,7 @@ type Banner = {
 
 - **Default:** `{}`
 
-Inject content into the top of each JS, CSS or DTS file.
+Inject content into the top of each JavaScript, CSS or DTS output file.
```
Comment (Timeless0911) [inline website/docs/en/config/lib/banner.mdx:19]: update the jsdoc at the same time

File: website/docs/en/config/lib/footer.mdx [hunk 2]
```diff
@@ -16,7 +16,7 @@ type Footer = {
 
 - **Default:** `{}`
 
-Inject content into the bottom of each JS, CSS or DTS file.
+Inject content into the bottom of each JavaScript, CSS or DTS file.
```
Comment (Timeless0911) [inline website/docs/en/config/lib/footer.mdx:19]: jsdoc

File: website/docs/zh/config/lib/auto-extension.mdx [hunk 3]
```diff
@@ -1 +1,52 @@
...
+
+举例来说，如果 `foo` 是一个文件夹，你需要将 `import * from './foo'` 改写为 `import * from './foo/index'`。
+
+:::
+
+当 `autoExtension` 设置为 `false` 时，文件扩展名将默认为 `.js`。
+
+## 自定义扩展名
+
+你可以将 `autoExtension` 设置为 `false`，并使用 [output.filename](https://rsbuild.dev/zh/config/output/filename) 来自定义 JavaScript 输出文件。
```
Comment (Timeless0911) [inline website/docs/zh/config/lib/auto-extension.mdx:28]:
```suggestion
你可以将 `autoExtension` 设置为 `false`，并使用 [output.filename](/config/rsbuild/output#outputfilename) 来自定义 JavaScript 输出文件。
```

File: website/docs/zh/config/lib/bundle.mdx [hunk 4]
```diff
@@ -1 +1,144 @@
 # lib.bundle
+
+- **类型：** `boolean`
+- **默认值：** `undefined`
+
+指定是否打包库，当 `bundle` 设置为 `true` 时称为打包模式，设置为 `false` 时称为 bundleless 模式。更多详情请参见 [bundle / bundleless](/guide/basic/output-structure#bundle--bundleless)。
```
Comment (Timeless0911) [inline website/docs/zh/config/lib/bundle.mdx:6]:
```suggestion
指定是否打包库，当 `bundle` 设置为 `true` 时称为 bundle 模式，设置为 `false` 时称为 bundleless 模式。更多详情请参见 [bundle / bundleless](/guide/basic/output-structure#bundle--bundleless)。
```

File: website/docs/zh/config/lib/bundle.mdx [hunk 5]
```diff
@@ -1 +1,144 @@
...
+
+::: warning
+无打包模式尚未完全支持，某些功能如 [资源](/guide/advanced/assets) 可能无法正常工作。
```
Comment (Timeless0911) [inline website/docs/zh/config/lib/bundle.mdx:9]:
```suggestion
bundleless 模式尚未完全支持，某些功能如 [资源](/guide/advanced/assets) 可能无法正常工作。
```

Comment (Timeless0911) [reply website/docs/zh/config/lib/bundle.mdx:9]: I remember I have remove the `/guide/advanced/assets` docs, maybe we should update here in both zh and en docs.

File: website/docs/zh/config/lib/bundle.mdx [hunk 6]
```diff
@@ -1 +1,144 @@
...
+:::
+
+## 设置入口
+
+我们需要为构建指定入口文件。
+
+### bundle: true
+
+当 `bundle` 设置为 `true` 时，入口需要设置为入口文件。默认入口为 `src/index.(ts|js|tsx|jsx|mjs|cjs)`。你需要确保入口文件存在，或通过 [source.entry](https://rsbuild.dev/zh/config/source/entry) 配置自定义入口。
```
Comment (Timeless0911) [inline website/docs/zh/config/lib/bundle.mdx:18]: do we need to update all rsbuild config links from `/config/rsbuild/...`?

File: website/docs/zh/config/lib/redirect.mdx [hunk 7]
```diff
@@ -1 +1,50 @@
...
+:::
+
+如果你不需要这些重定向，可以关闭它，这样导入路径将保持不变。
+
+```ts title="rslib.config.ts"
+export default {
+  lib: [
+    {
+      redirect: {
+        style: false, // Turn off the redirect of the style file
```
Comment (Timeless0911) [inline website/docs/zh/config/lib/redirect.mdx:36]:
```suggestion
        style: false, // 关闭样式文件的重定向
```

File: website/docs/zh/config/lib/syntax.mdx [hunk 8]
```diff
@@ -1 +1,69 @@
...
+  | 'esnext';
+
+type Syntax = EcmaScriptVersion | string[];
+```
+
+- **默认值：** `'esnext'`
+
+配置 JavaScript 和 CSS 将被降级的语法版本。
+
+参见 [输出兼容性 - 语法降级](/guide/advanced/output-compatibility) 了解更多详情。
```
Comment (Timeless0911) [inline website/docs/zh/config/lib/syntax.mdx:28]:
```suggestion
参见 [产物兼容性 - 语法降级](/guide/advanced/output-compatibility) 了解更多详情。
```

File: website/docs/zh/config/lib/bundle.mdx (hunk 6 above)
Comment (fi3ework) [reply website/docs/zh/config/lib/bundle.mdx:18]: I have considered doing this, but currently, Rslib's Rsbuild has very little content aside from the redirection. I think we can change the link after we add enough content.

File: website/docs/zh/config/lib/bundle.mdx (hunk 5 above)
Comment (fi3ework) [reply website/docs/zh/config/lib/bundle.mdx:9]: The asset page is available only by a link, https://lib.rsbuild.dev/guide/advanced/assets. I think it's fine to leave it there as a placeholder.

**Review by Timeless0911 — APPROVED** [review]

**Pull request merged**